Makefile

```diff
@@ -10,10 +10,24 @@
 # or export those into environment, or pass on the command line e.g.
 # make all DEBUG=1 CXX=g++-12
 
+HOST_OS = $(shell uname -s)
+
+ifeq ($(HOST_OS), Darwin)
+CXX = g++-14 # from homebrew, needed for 128-bit floats
```

Comment:
The Apple clang doesn't support C++-20 enough to compile filesystem operations. It doesn't have any 128-bit float, as far as I can tell. The real GCC resolves those issues.